restcomm/restcomm.mscontrol.mms/src/main/java/org/restcomm/connect/mscontrol/mms/MgcpMediaGroup.java

```diff
@@ -220,29 +232,28 @@ protected void play(final Object message) {
         final List<URI> uris = request.uris();
         final int iterations = request.iterations();
         final org.restcomm.connect.mgcp.Play play = new org.restcomm.connect.mgcp.Play(uris, iterations);
-        this.lastEvent = AUMgcpEvent.aupa;
         stop(lastEvent);
+        this.lastEvent = AUMgcpEvent.aupa;
         this.originator = sender();
         ivr.tell(play, self);
         ivrInUse = true;
     }
 
     @SuppressWarnings("unchecked")
     protected void notification(final Object message) {
-        final IvrEndpointResponse<String> response = (IvrEndpointResponse<String>) message;
+        final IvrEndpointResponse response = (IvrEndpointResponse) message;
         final ActorRef self = self();
-        MediaGroupResponse<String> event = null;
+        MediaGroupResponse<CollectedResult> event;
         if (response.succeeded()) {
-            event = new MediaGroupResponse<String>(response.get());
+            event = new MediaGroupResponse<>(response.get());
         } else {
-            event = new MediaGroupResponse<String>(response.cause(), response.error());
+            event = new MediaGroupResponse<>(response.cause(), response.error());
         }
-        // for (final ActorRef observer : observers) {
-        // observer.tell(event, self);
-        // }
         if (originator != null)
             this.originator.tell(event, self);
-        ivrInUse = false;
+        if (!response.get().isPartial()) {
```

Comment:
This is unsafe and can cause NPE

Here is a sample

*MediaServer logs*

```
11:25:22,508 DEBUG [SubMgcpTransactionManager] Closed transaction 147483695 for call agent /10.0.0.4:2727 with code 510
11:25:22,508 DEBUG [MgcpChannel] Outgoing MGCP message to /10.0.0.4:2727:

510 147483695 The transaction could not be executed, because some unspecified protocol error was detected
```

*Corresponding restcomm behaviour*

```
11:25:22,512 INFO  [org.restcomm.connect.mscontrol.mms.MgcpMediaGroup] (RestComm-akka.actor.default-dispatcher-150) ********** Media Group akka://RestComm/user/$h/$l/$a/$a Current State: "active
11:25:22,512 INFO  [org.restcomm.connect.mscontrol.mms.MgcpMediaGroup] (RestComm-akka.actor.default-dispatcher-150) ********** Media Group akka://RestComm/user/$h/$l/$a/$a Processing Message: "org.restcomm.connect.mgcp.IvrEndpointResponse sender : class akka.actor.LocalActorRef
11:25:22,512 INFO  [org.restcomm.connect.interpreter.VoiceInterpreter] (RestComm-akka.actor.default-dispatcher-150)  ********** VoiceInterpreter's akka://RestComm/user/$h/$k Current State: gathering
, Processing Message: org.restcomm.connect.mscontrol.api.messages.MediaGroupResponse
11:25:22,512 ERROR [org.restcomm.connect.commons.faulttolerance.RestcommSupervisorStrategy] (RestComm-akka.actor.default-dispatcher-153) RestcommSupervisorStrategy, actor exception handling. Actor path akka://RestComm/user/$h/$l/$a/$a, exception cause java.lang.NullPointerException, default exception handling strategy Resume
11:25:22,512 DEBUG [org.mobicents.servlet.sip.core.session.SipSessionImpl] (RestComm-akka.actor.default-dispatcher-154) dialog associated with this session to create the new request BYE within that dialog gov.nist.javax.sip.stack.SIPDialog@eebc2806
11:25:22,512 DEBUG [org.mobicents.servlet.sip.core.session.SipSessionImpl] (RestComm-akka.actor.default-dispatcher-154) dialog state Confirmed Dialog for that dialog gov.nist.javax.sip.stack.SIPDialog@eebc2806
11:25:22,512 ERROR [org.restcomm.connect.mscontrol.mms.MgcpMediaGroup] (RestComm-akka.actor.default-dispatcher-150) null: java.lang.NullPointerException
	at org.restcomm.connect.mscontrol.mms.MgcpMediaGroup.notification(MgcpMediaGroup.java:254) [restcomm-connect.mscontrol.mms-8.2.0.136.jar:8.2.0.136]
	at org.restcomm.connect.mscontrol.mms.MgcpMediaGroup.onReceive(MgcpMediaGroup.java:357) [restcomm-connect.mscontrol.mms-8.2.0.136.jar:8.2.0.136]
	at akka.actor.UntypedActor$$anonfun$receive$1.applyOrElse(UntypedActor.scala:159) [akka-actor_2.10-2.1.2.jar:]
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:425) [akka-actor_2.10-2.1.2.jar:]
	at akka.actor.ActorCell.invoke(ActorCell.scala:386) [akka-actor_2.10-2.1.2.jar:]
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:230) [akka-actor_2.10-2.1.2.jar:]
	at akka.dispatch.Mailbox.run(Mailbox.scala:212) [akka-actor_2.10-2.1.2.jar:]
	at akka.dispatch.ForkJoinExecutorConfigurator$MailboxExecutionTask.exec(AbstractDispatcher.scala:506) [akka-actor_2.10-2.1.2.jar:]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:262) [scala-library-2.10.1.jar:]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:975) [scala-library-2.10.1.jar:]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1478) [scala-library-2.10.1.jar:]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104) [scala-library-2.10.1.jar:]

```